PurchasesTests/Support/BeginRefundRequestHelperTests.swift

```diff
@@ -33,126 +98,232 @@ class BeginRefundRequestHelperTests: XCTestCase {
 
-        helper.beginRefundRequest(productID: mockProductID) { result in
-            callbackCalled = true
-            receivedResult = result
-        }
+        let receivedStatus = try await helper.beginRefundRequest(forProduct: mockProductID)
 
-        expect(callbackCalled).toEventually(beTrue())
-        let nonNilReceivedResult: Result<RefundRequestStatus, Error> = try XCTUnwrap(receivedResult)
         let expectedStatus = RefundRequestStatus.success
-        expect(nonNilReceivedResult).to(beSuccess { status in
-            expect(status) == expectedStatus
-        })
+        expect(receivedStatus) == expectedStatus
     }
 
-    func testBeginRefundReturnsFailureOnStoreKitRefundRequestFailure() throws {
+    func testBeginRefundReturnsFailureOnStoreKitRefundRequestFailure() async throws {
         guard #available(iOS 15.0, macCatalyst 15.0, *) else {
             throw XCTSkip("Required API is not available for this test.")
         }
 
         let expectedError = ErrorUtils.beginRefundRequestError(withMessage: "test")
         sk2Helper.maybeMockSK2Error = expectedError
 
-        var callbackCalled = false
-        var receivedResult: Result<RefundRequestStatus, Error>?
 
-        helper.beginRefundRequest(productID: mockProductID) { result in
-            callbackCalled = true
-            receivedResult = result
+        do {
+            _ = try await helper.beginRefundRequest(forProduct: mockProductID)
+            XCTFail("beginRefundRequestForProduct should have thrown error")
+        } catch {
+            expect(error).to(matchError(expectedError))
+        }
+    }
+
+    func testBeginRefundForEntitlementFailsOnCustomerInfoFetchFail() async throws {
+        guard #available(iOS 15.0, macCatalyst 15.0, *) else {
+            throw XCTSkip("Required API is not available for this test.")
+        }
+
+        customerInfoManager.stubbedError = ErrorUtils.customerInfoError(withMessage: "")
+
+        let expectedError = ErrorUtils.beginRefundRequestError(withMessage: Strings.purchase.begin_refund_customer_info_error(entitlementID: mockEntitlementID).description)
+
+        do {
+            _ = try await helper.beginRefundRequest(forEntitlement: mockEntitlementID)
+            XCTFail("beginRefundRequestForEntitlement should have thrown error")
+        } catch {
+            expect(error).to(matchError(expectedError))
+            expect(error.localizedDescription) == expectedError.localizedDescription
+        }
+    }
+
+    func testBeginRefundForActiveEntitlementFailsOnCustomerInfoFetchFail() async throws {
+        guard #available(iOS 15.0, macCatalyst 15.0, *) else {
+            throw XCTSkip("Required API is not available for this test.")
+        }
+
+        customerInfoManager.stubbedError = ErrorUtils.customerInfoError(withMessage: "")
+
+        let expectedError = ErrorUtils.beginRefundRequestError(withMessage: Strings.purchase.begin_refund_customer_info_error(entitlementID: nil).description)
```

Comment:
nitpick: add line breaks to prevent going over 120 chars

Reply:
ah shoot, it showed on the next line because my screen wasn't wide enough 🙄